src/functional/merge.js

```diff
@@ -38,7 +38,8 @@ export function mergeWithSources(collection, sources, merger) {
     );
   }
   if (isImmutable(collection)) {
-    return collection.mergeWith
+    const shouldMergeWith = collection.mergeWith && merger instanceof Function;
```

Comment:
This logic translates all calls to `Immutable.merge(collection, sources)` into calls to `collection.concat(...sources)`. Should instead check for `collection.merge` if `merger` isn't a function. Something like:

```javascript
if (collection.mergeWith && merger instanceOf Function) {
  return collection.mergeWith(merger, ...sources);
} else if (collection.merge && !(merger instanceOf Function) { // REALLY not sure about this condition
  return collection.merge(...sources);
} else {
  return collection.concat(...sources);
}